extensions/ql-vscode/src/vscode-tests/no-workspace/query-results.test.ts

```diff
@@ -216,7 +219,88 @@ describe('query-results', () => {
       },
       sourceInfo as SourceInfo
     );
-    expect(results3).to.deep.eq({ a: 6, t: 'SarifInterpretationData' });
+    // We do not re-interpret if we are reading from a SARIF file. 
+    expect(spy).to.not.have.been.called;
+
+    expect(results3).to.have.property('t', 'SarifInterpretationData');
+    expect(results3).to.have.nested.property('runs[0].results');
+
+    // try a fourth time, but this time we use an invalid small SARIF file
+    spy.reset();
+    fs.writeFileSync(interpretedResultsPath, JSON.stringify({
+      a: '6' // Invalid: no runs or results
+    }), 'utf8');
+
+    await expect(
+      interpretResultsSarif(
+        mockServer,
+        metadata,
+        {
+          resultsPath, interpretedResultsPath
+        },
+        sourceInfo as SourceInfo)
+    ).to.be.rejectedWith('Parsing output of interpretation failed: Invalid SARIF file: expecting at least one run with result.');
+
+    // We do not attempt to re-interpret if we are reading from a SARIF file. 
+    expect(spy).to.not.have.been.called;
+
+    // Try a fifth time with a valid large SARIF file
+    spy.reset();
+    const validSarifStream = fs.createWriteStream(interpretedResultsPath, { flags: 'w' });
+
+    validSarifStream.write(JSON.stringify({
+      runs: [{ results: [] }] // A run needs results to succeed.
+    }), 'utf8');
+
+    for (let i = 0; i < 10000000; i++) {
+      validSarifStream.write(JSON.stringify({
+        a: '6'
+      }), 'utf8');
+    }
+    validSarifStream.end();
+
+    const results5 = await interpretResultsSarif(
+      mockServer,
+      metadata,
+      {
+        resultsPath, interpretedResultsPath
+      },
+      sourceInfo as SourceInfo
+    );
+    // We do not re-interpret if we are reading from a SARIF file. 
+    expect(spy).to.not.have.been.called;
+
+    expect(results5).to.have.property('t', 'SarifInterpretationData');
+    expect(results5).to.have.nested.property('runs[0].results');
+
+    // Explicitly delete the large SARIF file — overwriting causes odd errors.
+    fs.unlink(interpretedResultsPath, (err) => {
+      if (err) {
+        throw err;
+      }
+    });
+
+    // Try a sixth time with an invalid large SARIF file
+    spy.reset();
+    const invalidSarifStream = fs.createWriteStream(interpretedResultsPath, { flags: 'w' });
+    for (let i = 0; i < 10000000; i++) {
+      invalidSarifStream.write(JSON.stringify({
+        a: '6'
+      }), 'utf8');
+    }
+
+    await expect(
+      interpretResultsSarif(
+        mockServer,
+        metadata,
+        {
+          resultsPath, interpretedResultsPath
+        },
+        sourceInfo as SourceInfo)
+    ).to.be.rejectedWith('Parsing output of interpretation failed: Invalid SARIF file: expecting at least one run with result.');
```

Comment:
This invalid large SARIF file test is not rejecting as expected from the parser (which is what we'd expect, similar to the small invalid file), but rather returning empty/undefined and `interpretResultsSarif` is appending and returning `{ t: 'SarifInterpretationData' }` alone. Will investigate further.

Reply:
Hm.. it may still be related to some issue with the `WriteStream`. When I only write a single object to the file, which should make it identical to test 4, I still get this result rather than the error thrown.

```
    for (let i = 0; i < 1; i++) {
      invalidSarifStream.write(JSON.stringify({
        a: '6'
      }), 'utf8');
    }
```

(using this loop)

Reply:
😕

Reply:
It's really odd, I had thought deleting the file resolved this issue, but it just creates a different error (not throwing any parsing error at all rather than throwing unexpected parsing errors). But it's definitely something related to file system writing 🤔